lib/widgets/message_list.dart

```diff
@@ -56,21 +57,32 @@ class _MessageListPageState extends State<MessageListPage> {
   Widget build(BuildContext context) {
     final store = PerAccountStoreWidget.of(context);
 
-    final Color backgroundColor;
+    final Color? backgroundColor;
+    bool removeAppBarBottomBorder = false;
     switch(widget.narrow) {
       case AllMessagesNarrow():
-        backgroundColor = _kFallbackStreamColor;
+        backgroundColor = null; // i.e., inherit
       case StreamNarrow(:final streamId):
       case TopicNarrow(:final streamId):
         backgroundColor = store.subscriptions[streamId]?.colorSwatch().barBackground
-          ?? _kFallbackStreamColor;
+          ?? _kUnsubscribedStreamRecipientHeaderColor;
+        // All recipient headers will match this color; remove distracting line
+        // (but are recipient headers even needed for topic narrows?)
+        removeAppBarBottomBorder = true;
       case DmNarrow():
```

Comment:
nit: separate the cases (here and above) visually, now that they're longer:
```suggestion
        removeAppBarBottomBorder = true;

      case DmNarrow():
```